File: packages/cexplorer-api/src/endpoints/wallet.ts
import type { CompareWalletsResponse } from "@/types/walletTypes";

import { handleFetch } from "@/lib/handleFetch";

/**
 * Fetches detailed comparison information for wallets from the articles endpoint.
 *
 * This request fetches a specific article page with the URL "wallets" in English.
 *
 * @returns {Promise<CompareWalletsResponse>} A promise resolving to the wallet comparison data.
 *
 * @throws Will throw an error if the fetch request fails.
 */
export const compareWallets = async () => {
  const url = `/article/detail`;
  const options = {
    params: {
      lng: "en",
      type: "page",
      url: "wallets",
    },
  };

  return handleFetch<CompareWalletsResponse>(url, undefined, options);
};

File: packages/cexplorer-api-rs/src/endpoints/wallet.rs
use crate::client::fetch_with_params;
use crate::error::CexplorerError;
use crate::types::wallet_types::*;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct CompareWalletsParams {
    pub lng: String,
    #[serde(rename = "type")]
    pub wallet_type: String,
    pub url: String,
}

pub async fn compare_wallets() -> Result<CompareWalletsResponse, CexplorerError> {
    let endpoint = "/article/detail";
    let params = CompareWalletsParams {
        lng: "en".to_string(),
        wallet_type: "page".to_string(),
        url: "wallets".to_string(),
    };
    fetch_with_params::<CompareWalletsResponse, CompareWalletsParams>(endpoint, Some(&params)).await
}
